src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleShell.cs

```diff
@@ -33,14 +33,21 @@ public static int Start(string? bannerText, string? helpText, string?[] args)
         /// <returns>An integer value which should be used as exit code for the process.</returns>
         public static int Start(InitialSessionState initialSessionState, string? bannerText, string? helpText, string?[] args)
         {
-            if (initialSessionState == null)
+            if (args == null)
             {
-                throw PSTraceSource.NewArgumentNullException(nameof(initialSessionState));
+                throw PSTraceSource.NewArgumentNullException(nameof(args));
             }
 
-            if (args == null)
+            ConsoleHost.ParseCommandLine(args);
+
+            return InternalStart(initialSessionState, bannerText, helpText, args);
+        }
+
+        internal static int InternalStart(InitialSessionState initialSessionState, string? bannerText, string? helpText, string?[] args)
```

Comment:
Why is `InternalStart` needed?

Reply:
This addresses your comment https://github.com/PowerShell/PowerShell/pull/11482#discussion_r467365335
There was two overlapped code paths with two conflicts - (1) call parser, (2) initialSessionState. So I had to add the method to resolve the conflicts and address your comment.

Reply:
Done.